hack/kind-up.sh

```diff
@@ -118,6 +118,38 @@ setup_loopback_device() {
   echo "Setting up loopback device ${LOOPBACK_DEVICE} completed."
 }
 
+# setup_containerd_registry_mirrors sets up all containerd registry mirrors.
+# Resources:
+# - https://github.com/containerd/containerd/blob/main/docs/hosts.md
+# - https://kind.sigs.k8s.io/docs/user/local-registry/
+setup_containerd_registry_mirrors() {
+  for node in $(kind get nodes --name="$CLUSTER_NAME"); do
+    setup_containerd_registry_mirror "localhost:5001" "http://localhost:5001" 5001
+    setup_containerd_registry_mirror "gcr.io" "https://gcr.io" 5003
+    setup_containerd_registry_mirror "eu.gcr.io" "https://eu.gcr.io" 5004
+    setup_containerd_registry_mirror "ghcr.io" "https://ghcr.io" 5005
+    setup_containerd_registry_mirror "registry.k8s.io" "https://registry.k8s.io" 5006
+    setup_containerd_registry_mirror "quay.io" "https://quay.io" 5007
+  done
+}
+
+# setup_containerd_registry_mirror sets up a given contained registry mirror.
+setup_containerd_registry_mirror() {
+  host=$1
+  server=$2
+  mirror_port=$3
+
+  echo "Setting up containerd registry mirror for host ${host}.";
+  registry_name="garden.local.gardener.cloud"
```

Comment:
Previously, the registry name came from `.Values.registry.hostname`, which is different for different scenarios, e.g. simple local kind vs. gardener operator vs. extension scenario. Could you please explain why this would work in all scenarios?

Reply:
You are right, let me check this.

Reply:
I now tested the operator local setup and `garden.local.gardener.cloud` indeed works. Currently in the master branch, we pass the control-plane node name as hostname to the registry.
https://github.com/gardener/gardener/blob/a89ca663f0ba4c6f576db6f06258e627bd5c56f4/example/gardener-local/kind/operator/values.yaml#L9-L10

3months ago you adapted coredns config with the following change:
https://github.com/gardener/gardener/blob/50a5a60b85bd908d28caabfa5ca3fa504057ee21/hack/kind-up.sh#L247-L266

If I am not wrong, this resolves `garden.local.gardener.cloud` always to the IP of the control plane node.
If that's the case, I assume we can drop passing the custom hostnames for the registry and always use `garden.local.gardener.cloud` in the contrainerd config. WDYT?

Reply:
> I assume we can drop passing the custom hostnames for the registry and always use `garden.local.gardener.cloud` in the contrainerd config. WDYT?

That sounds good. When we introduced `garden.local.gardener.cloud`, this was exactly the motivation: to have a name pointing to the control plane node that works everywhere (i.e., from within seeds and shoots, for IPv4 and IPv6). So it makes perfect sense to use it for the mirror config as well.

Reply:
Thanks for the confirmation of our assumptions! 

Several hours ago we had a short call with @ScheererJ and we agreed to use `garden.local.gardener.cloud` in unified manner. I created a dedicated PR for it - https://github.com/gardener/gardener/pull/8063. I will adapt this one, once/if https://github.com/gardener/gardener/pull/8063 gets merged.